frontend/src/components/ActivityReportsTable/index.js

```diff
@@ -0,0 +1,339 @@
+  useEffect(() => {
+    setReportCheckboxes(makeReportCheckboxes(reports, false));
+  }, [reports]);
+
+  useEffect(() => {
+    const checkValues = Object.values(reportCheckboxes);
+    if (checkValues.every((v) => v === true)) {
+      setAllReportsChecked(true);
+    } else if (allReportsChecked === true) {
+      setAllReportsChecked(false);
+    }
+  }, [reportCheckboxes, allReportsChecked]);
+
+  const handleReportSelect = (event) => {
+    const { target: { checked = null, value = null } = {} } = event;
+    if (checked === true) {
+      setReportCheckboxes({ ...reportCheckboxes, [value]: true });
+    } else {
+      setReportCheckboxes({ ...reportCheckboxes, [value]: false });
+    }
+  };
+
+  const handlePageChange = (pageNumber) => {
+    if (!loading) {
+      setActivePage(pageNumber);
+      setOffset((pageNumber - 1) * perPage);
+    }
+  };
+
+  const requestSort = (sortBy) => {
+    let direction = 'asc';
+    if (
+      sortConfig
+      && sortConfig.sortBy === sortBy
+      && sortConfig.direction === 'asc'
+    ) {
+      direction = 'desc';
+    }
+    setActivePage(1);
+    setOffset(0);
+    setSortConfig({ sortBy, direction });
+  };
+
+  const handleDownloadAllReports = () => {
+    const filterQuery = filtersToQueryString(filters);
+    const downloadURL = getAllReportsDownloadURL(filterQuery);
+    window.location.assign(downloadURL);
+  };
+
+  const handleDownloadClick = () => {
+    const toDownloadableReportIds = (accumulator, entry) => {
+      if (!reports) return accumulator;
+      const [key, value] = entry;
+      if (value === false) return accumulator;
+      accumulator.push(key);
+      return accumulator;
+    };
+
+    const downloadable = Object.entries(reportCheckboxes).reduce(toDownloadableReportIds, []);
+    if (downloadable.length) {
+      const downloadURL = getReportsDownloadURL(downloadable);
+      window.location.assign(downloadURL);
+    }
+  };
+
+  const getClassNamesFor = (name) => (sortConfig.sortBy === name ? sortConfig.direction : '');
+  const renderColumnHeader = (displayName, name) => {
+    const sortClassName = getClassNamesFor(name);
+    let fullAriaSort;
+    switch (sortClassName) {
+      case 'asc':
+        fullAriaSort = 'ascending';
+        break;
+      case 'desc':
+        fullAriaSort = 'descending';
+        break;
+      default:
+        fullAriaSort = 'none';
+        break;
+    }
+    return (
+      <th scope="col" aria-sort={fullAriaSort}>
+        <a
+          role="button"
+          tabIndex={0}
+          onClick={() => {
+            requestSort(name);
+          }}
+          onKeyPress={() => requestSort(name)}
+          className={`sortable ${sortClassName}`}
+          aria-label={`${displayName}. Activate to sort ${sortClassName === 'asc' ? 'descending' : 'ascending'
+          }`}
+        >
+          {displayName}
+        </a>
+      </th>
+    );
+  };
+
+  const displayReports = reports.length ? reports : [emptyReport];
```

Comment:
I'm probably not understanding something - do we want an row with an empty report rather than an empty array?

Reply:
I'm not too sure, I was trying to move over the table to it's own component without changing how the report works in any way. Now that it is in it's own component it is easier to refactor.